github/open_close_pr.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/python3
+
+import github3
+import argparse
+import logging
+import yaml
+from cachecontrol.adapter import CacheControlAdapter
+
+
+REF_FORMAT = 'refs/heads/'
+DEFAULT_COMMIT_MSG = 'automated commit'
+FREEIPA_PRCI_CONFIG_FILE = '.freeipa-pr-ci.yaml'
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+consoleHandler = logging.StreamHandler()
+consoleHandler.setLevel(logging.INFO)
+logger.addHandler(consoleHandler)
+
+
+def load_yaml(yml_path):
+    try:
+        with open(yml_path) as yml_file:
+            return yaml.load(yml_file)
+    except IOError as exc:
+        raise argparse.ArgumentTypeError(
+            'Failed to open {}: {}'.format(yml_path, exc))
+    except yaml.YAMLError as exc:
+        raise argparse.ArgumentTypeError(
+            'Failed to parse YAML from {}: {}'.format(yml_path, exc))
+
+
+class AutomatedPR(object):
+
+    def __init__(self, github_token, repo):
+        github = github3.login(token=github_token)
+        github.session.mount('https://api.github.com',
+                             CacheControlAdapter())
+
+        self.repo = github.repository(repo['owner'], repo['name'])
+
+    def create_branch(self, identifier):
+        last_commit = next(self.repo.commits())
+        new_branch_ref = '{}{}'.format(REF_FORMAT, identifier)
+        self.repo.create_ref(new_branch_ref, last_commit.sha)
+
+        return identifier
+
+    def commit_new_prci_config_file(self, new_prci_config, branch_name):
+        """
+        Updates the .freeipa-pr-ci.yaml file with the content
+        of the file provided.
+        """
+        prci_config_file = self.repo.file_contents(FREEIPA_PRCI_CONFIG_FILE)
+
+        content = ''
+        with open(new_prci_config, 'r') as f:
+            content = f.read()
+
+        prci_config_file.update(DEFAULT_COMMIT_MSG, content.encode(),
+                                branch_name)
+
+    def close_older_pr(self, pr_identifier):
+        refs = {r.ref:r for r in self.repo.refs()}
+
+        for pull_request in self.repo.pull_requests():
+            if '[{}]'.format(pr_identifier) in pull_request.title:
+                ref_uri = '{}{}'.format(REF_FORMAT, pull_request.head.ref)
+                ref = refs[ref_uri]
+                ref.delete()
+                logger.debug("PR #%s closed and branch deleted",
+                             pull_request.number)
+
+    def rebase_branch(self, base_branch):
```

Comment:
yeah, I don't like either. However, I don't think it's possible to do the behavior above using the github3 lib. Probably, it would be necessary to use `Popen` and call the commands by ourselves. It's not a big deal anyway.